### ISSUE TITLE
How to setup prepared_max on connection

### ISSUE THREAD

**Issue by MRigal** [psycopg/psycopg#935]
Hi,

Using sqlalchemy + psycopg + pgBouncer without libpq>=17, I was encountering rare but regular `ProgrammingError: (psycopg.errors.InvalidSqlStatementName) prepared statement "_pg3_XXX" does not exist`.

Then I found the hint to set `prepared_max` to None in https://www.psycopg.org/psycopg3/docs/advanced/prepare.html#prepared-statements.

Ideally, I would like to pass it to the connection builder of sqlalchemy

This would translate to `psycopg.Connection.connect(dsn, prepared_max=None)`.

However, `prepared_max` is not handled by the `connect` classmethod https://github.com/psycopg/psycopg/blob/master/psycopg/psycopg/connection.py#L80 thus resulting in the parameter not being set at connection.

I can set it up using events in sqlalchemy, here is my snippet:
```
from sqlalchemy import Pool
from sqlalchemy.event import listens_for
from sqlalchemy.pool.base import _ConnectionRecord

@listens_for(Pool, "connect")
def _on_connect_set_prepared_max(dbapi_con: Connection, connection_record: _ConnectionRecord) -> None:
    dbapi_con.prepared_max = None
```

However I wonder if it was really planned that it is not possible to set it up at connection time. Is it so? It fails silently and I was not expecting that. If desired, I could make a PR to handle it at `connect()` level.

Thanks in advance

**Comment by dvarrazzo** [comment]
You don't want to se t `prepared_max = None` because that means "prepare an unbounded number of statements". You want to set `prepared_threshold = None`, and the very page you linked says:

> You can disable the use of prepared statements on a connection by setting its [prepare_threshold](https://www.psycopg.org/psycopg3/docs/api/connections.html#psycopg.Connection.prepare_threshold) attribute to None.
>
> Changed in version 3.1: You can set prepare_threshold as a [connect()](https://www.psycopg.org/psycopg3/docs/api/connections.html#psycopg.Connection.connect) keyword parameter too.

So I believe you have everything you need.

**Comment by MRigal** [comment]
@dvarrazzo I was following the later hint on the same page

> If libpq 17 is not available on your client, but PgBouncer is 1.22 or higher, you can still use Psycopg as long as you disable deallocation.
> 
> You can do so by setting [Connection.prepared_max](https://www.psycopg.org/psycopg3/docs/api/connections.html#psycopg.Connection.prepared_max) to None.

Is it that bad to have an unbounded number of statements? Can it go up to memory overflow?

Else I can set `prepare_threshold` to None, but I was hoping to be able to still use prepared statements...
